nilearn/plotting/displays.py

```diff
@@ -137,6 +142,75 @@ def draw_left_right(self, size, bg_color, **kwargs):
                 bbox=dict(boxstyle="square,pad=0", ec=bg_color, fc=bg_color),
                 **kwargs)
 
+    def draw_scale_bar(self, bg_color, scalebar_kwargs, **kwargs):
+        """ Adds a scale bar annotation to the display
+
+        Parameters
+        ----------
+
+        bgcolor: matplotlib color: string or (r, g, b) value
+            The background color of the scale bar annotation
+        scalebar_kwargs: dict
+            Arguments to define how the scale bar will be rendered.
+            Allowed keys are:
+            `scale_width` (a float with default: 5.0
+            that defines the width of the scale given in `units`);
+            `units` (str representing the physical units to use on
+            the scale bar -`'cm'` or `'mm'`);
+            `fontproperties` (a matplotlib's ``FontProperties`` object
+            or dict, if `fontproperties` is defined then the `size`
+            keyword argument will be dismissed);
+            `frameon` (boolean, whether the scale bar is plotted with
+            a border);
+            `loc` (matplotlib's location int or str indicating where
+            the scale bar should be placed with default: 4 - bottom-right);
+            `pad` (float - margin from the view edges);
+            `borderpad` (float - inner padding from the edges of the scale
+            bar annotation);
+            `sep` (float - separation between text and the bar);
+            for additional keywords please check matplotlib's
+            `AnchoredSizeBar <https://matplotlib.org/mpl_toolkits/axes_grid\
+            /api/anchored_artists_api.html>`__ documentation.
+        kwargs:
+            Extra keyword arguments are passed to matplotlib's
+            FontProperties object used by AnchoredSizeBar.
+        """
+        from mpl_toolkits.axes_grid1.anchored_artists import AnchoredSizeBar
+
+        # Sanity check to replace None by empty dict
+        scalebar_kwargs = scalebar_kwargs or {}
+
+        axis = self.ax
+
+        color = kwargs.pop('color')
+        font_properties = scalebar_kwargs.get('fontproperties',
+                                              FontProperties(**kwargs))
```

Comment:
This dance of kwargs makes me nervous. It's really hard to follow where arguments are routed even when reading the code. Hence for an end user, it is going to be guesswork.